### PR TITLE
WooCommerce Analytics: support product bundles add to carts

### PR DESCRIPTION
**Pull request by greenafrican**
IN PROGRESS

Fixes #

#### Changes proposed in this Pull Request:

* 

#### Testing instructions:

* Add WooCommerce Product Bundles plugin and create at least one product bundle as per instructions here - https://docs.woocommerce.com/document/bundles/bundles-configuration/#getting-started
* Make sure the carts events fire for each product in the bundle both from the lists. You need to go to the cart after adding it and should see a `product_view` and an `add_to_cart` event for each product and the bundle.
* Make sure the carts events fire for each product in the bundle from the product bundle page. You should see an `add_to_cart` event for each product and the bundle.

### CODE REVIEW

**Comment by psealock** [comment]
Tested the branch locally, I have one note.

Adding a bundle with two products creates 6 events
* `view` product 1
* `add_to_cart` product 1
* `view` product 2
* `add_to_cart` product 2
* `view` Bundle
* `add_to_cart` Bundle

I would have expected 4, but continuing through the checkout process produces 3 events for checkout and purchase, which makes sense. Tracking the bundle is valuable, but are we inflating events by including the bundle purchase?

File: modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php [hunk 1]
```diff
@@ -33,14 +33,15 @@ public function __construct() {
...
 		add_action( 'woocommerce_add_to_cart', array( $this, 'capture_add_to_cart' ), 10, 6 );
+		add_action( 'woocommerce_bundled_item_before_add_to_cart', array( $this, 'capture_bundle_add_to_cart' ), 10, 6 );
 
 		// single product page view
 		add_action( 'woocommerce_after_single_product', array( $this, 'capture_product_view' ) );
 
 		add_action( 'woocommerce_after_cart', array( $this, 'remove_from_cart' ) );
 		add_action( 'woocommerce_after_mini_cart', array( $this, 'remove_from_cart' ) );
 		add_action( 'wcct_before_cart_widget', array( $this, 'remove_from_cart' ) );
-		add_filter( 'woocommerce_cart_item_remove_link', array( $this, 'remove_from_cart_attributes' ), 10, 2 );
```
Comment (psealock) [inline modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php]: Curious here why this change?

Comment (greenafrican) [reply modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php]: That is an oversight on my part. I was testing `woocommerce_cart_item_remove` and forgot to change it back before committing. Thanks.

**Comment by greenafrican** [comment]
> I would have expected 4, but continuing through the checkout process produces 3 events for checkout and purchase, which makes sense. Tracking the bundle is valuable, but are we inflating events by including the bundle purchase?

Great question and I have been thinking about this too. My initial feeling is that it is ok as long as it doesn't duplicate sales figures. I think it would be useful to view the conversion of both the bundle and the bundle items.

Alternatively, what do you think about only tracking the bundle? I.E only send 1 event at checkout and purchase?

File: modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php [hunk 2]
```diff
@@ -158,6 +161,19 @@ public function remove_from_cart_attributes( $url, $key ) {
 		return $url;
 	}
 
+	/**
+	 * Queue remove from cart events for each item in a bundle
```
[the remark this answers is not in the document]
Comment (psealock) [reply modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php]: This is called on the `woocommerce_cart_item_removed` action. Would that not be called for every removal, not just ones involved in bundles?

[the remark this answers is not in the document]
Comment (psealock) [reply modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php]: I pulled this down to investigate. On a simple product removal I see `remove_from_cart` function appropriately firing. Then I see the event being passed to the session as well, but no event is fired on subsequent page reload. I'm not entirely sure why that is. 

Nonetheless can we differentiate between bundles and simple products in this function?

**Comment by brbrr** [comment]
Hey @greenafrican 
Pinging you in case you missed the reviews.
